cfgmgr/vxlanmgr.cpp

```diff
@@ -585,10 +879,168 @@ bool VxlanMgr::deleteVxlan(const VxlanInfo & info)
+    // Case 3: Entry exist but modified - Not taken care. Will address later
+     
+    if (true == m_in_reconcile)
+    { 
+        auto it = m_vxlanNetDevices.find(vxlan_dev_name);
+        if (it != m_vxlanNetDevices.end())
+        {
+            m_vxlanNetDevices.erase(it);
+            SWSS_LOG_NOTICE("Reconcile VxlanNetDevice %s reconciled. Pending %lu",
+                            vxlan_dev_name.c_str(), m_vxlanNetDevices.size());
+            return 0;
+        }
+        else
+        {
+            SWSS_LOG_NOTICE("Reconcile VxlanNetDevice %s doesn't not exist. Pending %lu",
+                            vxlan_dev_name.c_str(), m_vxlanNetDevices.size());
+        }
+    }
+    else
+    {
+        SWSS_LOG_NOTICE("Creating VxlanNetDevice %s", vxlan_dev_name.c_str());
+    }
+
+    // Remove kernel device if it exists
+    chkcmd = "ip link show " + vxlan_dev_name;
+    ret = swss::exec(chkcmd,res);
+    if(ret == 0)
+    {
+        cmds = "ip link del dev " + vxlan_dev_name ;
+        EXEC_WITH_ERROR_THROW(cmds, res);
+    }
+
+#if 0
+    cmds = std::string("")
+        + BASH_CMD + " -c \""
+        + IP_CMD + " link add " + vxlan_dev_name + " type vxlan id " + std::string(vni_id) + 
+        + " local " + src_ip + ((dst_ip  == "")? "":(" remote " + dst_ip)) + " nolearning " +" dstport 4789 " + " && "
+        + IP_CMD + " link set " + vxlan_dev_name + " master Bridge " 
+        + " && "
+        + BRIDGE_CMD + " vlan add vid " + std::string(vlan_id) + " dev " + vxlan_dev_name 
+        + " && "
+        + BRIDGE_CMD + " vlan add vid " + std::string(vlan_id) + " untagged pvid dev " + vxlan_dev_name 
+        + " && ";
+#endif
+
+    // ip link add <vxlan_dev_name> type vxlan id <vni> local <src_ip> remote <dst_ip> 
+    // dstport 4789
+    // ip link set <vxlan_dev_name> master DOT1Q_BRIDGE_NAME
+    // bridge vlan add vid <vlan_id> dev <vxlan_dev_name>
+    // bridge vlan add vid <vlan_id> untagged pvid dev <vxlan_dev_name>
+    // ip link set <vxlan_dev_name> up
+
+    link_add_cmd = std::string("") + IP_CMD + " link add " + vxlan_dev_name + 
+                   " address " + gMacAddress.to_string() + " type vxlan id " + 
+                   std::string(vni_id) + " local " + src_ip + 
+                   ((dst_ip  == "")? "":(" remote " + dst_ip)) + 
+                   " nolearning " + " dstport 4789 ";
+    
+    link_set_master_cmd = std::string("") + IP_CMD + " link set " + 
+                          vxlan_dev_name + " master Bridge ";
+
+    link_up_cmd = std::string("") + IP_CMD + " link set " + vxlan_dev_name + " up ";
+
+    bridge_add_cmd = std::string("") + BRIDGE_CMD + " vlan add vid " + 
+                     std::string(vlan_id) + " dev " + vxlan_dev_name;
+
+    bridge_untagged_add_cmd = std::string("") + BRIDGE_CMD + " vlan add vid " + 
+                              std::string(vlan_id) + " untagged pvid dev " + vxlan_dev_name;
+
+    bridge_del_vid_cmd = std::string("") + BRIDGE_CMD + " vlan del vid 1 dev " + 
```

Comment:
After the command ip link set <vxlan_dev_name> master DOT1Q_BRIDGE_NAME the vxlan dev automatically 
gets added to vid 1.. We wanted to avoid flooding vid-1 packets to vxlan device unless explicitly added. 

ip link add vtep1-22 type vxlan id 100 local 1.1.1.1 dstport 4789
ip link set vtep1-22 master Bridge

root@sonic:/home/admin# bridge vlan show
port    vlan ids
docker0  1 PVID Egress Untagged

Bridge  None
dummy    1 PVID Egress Untagged

vtep1-22         1 PVID Egress Untagged